File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/di/UserModule.java
package com.example.tutorial3scope1.di;


import com.example.tutorial3scope1.model.User;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class UserModule {

    @ActivityScope
    @Provides
    User provideUser() {
        return new User();
    }
}

File: Tutorial3-1Scope1/src/main/java/com/example/tutorial3scope1/di/UserModule.kt
package com.example.tutorial3scope1.di

import com.example.tutorial3scope1.model.User
import dagger.Module
import dagger.Provides

/**
 * @Provides method of module should have same scope with component, or both component and module
 * shouldn't have one.
 *
 * If it has a gloabl/application level scope then it's generated as Singleton object.
 * Without scope even it's a ApplicationModule objects are re-created
 */
@Module
class UserModule {

    @ActivityScope
    @Provides
    fun provideUser(): User {
        return User()
    }

}